presto-orc/src/main/java/com/facebook/presto/orc/reader/AbstractLongSelectiveStreamReader.java

```diff
@@ -206,4 +275,31 @@ protected void ensureOutputPositionsCapacity(int capacity)
             outputPositions = new int[capacity];
         }
     }
+
+    private void compactValues(int[] positions, int positionCount, boolean compactNulls)
+    {
+        int positionIndex = 0;
+        int nextPosition = positions[positionIndex];
+        for (int i = 0; i < outputPositionCount; i++) {
+            if (outputPositions[i] < nextPosition) {
+                continue;
+            }
+
+            assert outputPositions[i] == nextPosition;
```

Comment:
why don't we use `checkState`?

Reply:
this is performance critical code, hence, I'd like to avoid a check for each position